http/src/request/guild/member/search_guild_members.rs

```diff
@@ -0,0 +1,156 @@
+use crate::request::prelude::*;
+use bytes::Bytes;
+use serde::de::DeserializeSeed;
+use std::{
+    error::Error,
+    fmt::{Display, Formatter, Result as FmtResult},
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use twilight_model::{
+    guild::member::{Member, MemberDeserializer},
+    id::GuildId,
+};
+
+#[cfg(not(feature = "simd-json"))]
+use serde_json::Value;
+#[cfg(feature = "simd-json")]
+use simd_json::value::OwnedValue as Value;
+
+/// The error created when the members can not be queried as configured.
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub enum SearchGuildMembersError {
+    /// The limit is either 0 or more than 1000.
+    LimitInvalid {
+        /// Provided limit.
+        limit: u64,
+    },
+}
+
+impl Display for SearchGuildMembersError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        match self {
+            Self::LimitInvalid { .. } => f.write_str("the limit is invalid"),
+        }
+    }
+}
+
+impl Error for SearchGuildMembersError {}
+
+struct SearchGuildMembersFields {
+    query: String,
+    limit: Option<u64>,
+}
+
+/// Search the members of a specific guild by a query. Requires the [`GUILD_MEMBERS`] intent.
```

Comment:
this is no longer accurate, the members intent requirement for a query is a bug and should be fixed in the next discord api deploy according to staff on ddevs